src/mococrw/symmetric_memory.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <queue>
+
+namespace mococrw
+{
+
+class CipherMemoryStrategyI {
+public:
+    virtual ~CipherMemoryStrategyI() = default;
+
+    virtual void write(const std::vector<uint8_t> &&block) = 0;
```

Comment:
Any particular reason why you declare this as const when you require a &&-type ref? Also, should we just accept a plain `std::vector<uint8_t>` here?

Reply:
Done